### PR TITLE
go.mod: bump btclog dep

### PR DESCRIPTION
**Pull request by ellemouton**
Point to the latest version of the [v2 lib](https://github.com/btcsuite/btclog/pull/19) which checks log level before calling fmt.Sprint along
with a variety of other performance improvements.

h/t to @bitromortac for finding this! 🙏 

NOTE: This has been updated to remove the PrefixLogger as this can now be done via the btclog.Logger's WIthPrefix method

### CODE REVIEW

**Review by guggero — APPROVED** [review]
Nice, LGTM :tada:

File: build/prefix_log.go [hunk 1]
```diff
@@ -1,180 +0,0 @@
-package build
```
Comment (bitromortac) [inline build/prefix_log.go:1]: nice removal :tada:

**Review by bitromortac — APPROVED** [review]
LGTM :zap:

**Comment by ellemouton** [comment]
thanks for the reviews yall! think it needs an override merge cause of the required `unit-cover` test 🙏

**Pull request merged**